docs/SECURITY.md

```diff
@@ -0,0 +1,81 @@
+
+In short, EVE's deployment model forces us to protect against an opportunistic adversary with full physical access to and Edge Node and Edge Node's networks through a combination of system hardening, defense in depth, process isolation, secure autoupdate, measured boot, encryption, and seamless integration with EVE's controller security policies.
+
+## Security capabilities vs. security policies
+
+At the end of the day, EVE only does what its controller instructs it to do. This puts an enormous premium on protecting the trust between EVE instance and its controller at all costs (in both direction) but it also allows us to have a clear division of responsibilities between the two:
+
+* EVE only implements security capabilities, never security policies
+* Controller orchestrates EVE's security capabilities in order to realize sophisticated policies
+
+On the EVE's side, this separation allows us to focus on a small number of well designed capabilities (AKA security building blocks) that are guaranteed to have a well defined interactions between each other. For example, while EVE may provide a capability for storing encrypted binary blobs on the Edge Nodes it will be up to the controller's security policy to determine which pieces of data will be stored there.
+
+Controller and its management of security policies are out of scope for this document.
+
+## EVE threat model
+
+Borrowing a great deal from Chromium effort we consider two different kinds of adversaries:
+
+* An opportunistic adversary
+* A dedicated adversary
+
+The opportunistic adversary is just trying to compromise an individual Edge Node and/or data.  They are not targeting a specific user or enterprise, and they are not going to steal, disassemble or modify the Edge Node but they are very likely to have prolonged physical access to it. This level of physical access allows opportunistic adversary a chance to replace trusted network and I/O connections with those of their own making, though. Unlike Chromium, we assume that this will facilitate things like DNS or other network-level attacks and will place them within an opportunistic adversary's reach.
+
+The dedicated adversary may target a user or an enterprise specifically for attack.  They are willing to steal Edge Nodes to recover data or account credentials (not just to re-sell the device to make money). They are also willing and capable of modifying an Edge Node with extra hardware and software components. They may also do anything that the opportunistic adversary can do.
+
+For now we are mainly focusing on dangers posed by opportunistic adversaries.
```

Comment:
We should explain why.

Reply:
Well, the only explanation is lack of resources -- but I don't think it should be in the doc ;-)

Reply:
Actually, it is fair to say something like, "the EVE contributors and community need to prioritize which dangers/risks/etc. to focus on and in which order. For now, we are focusing mainly on dangers posed by opportunistic adversaries. As the the risk mitigation solidifies / project matures / contributors come on board / whatever, we will add focus to etc. etc."

Reply:
See the latest revision.